src/Strategy/HttpClients.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Http\Discovery\Strategy;
+
+use Http\Adapter\Guzzle6\Client as Guzzle6;
+use Http\Adapter\Guzzle5\Client as Guzzle5;
+use Http\Client\Curl\Client as Curl;
+use Http\Client\Socket\Client as Socket;
+use Http\Adapter\React\Client as React;
+use Http\Adapter\Buzz\Client as Buzz;
+
+/**
+ * Find common HTTP clients.
+ *
+ * @internal
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class HttpClients extends AbstractStrategy
+{
+    /**
+     * @var array
+     */
+    protected static $classes = [
+        'Http\Client\HttpAsyncClient' => [
+            ['class' => Guzzle6::class, 'condition' => Guzzle6::class],
+            ['class' => Curl::class, 'condition' => Curl::class],
+            ['class' => React::class, 'condition' => React::class],
+        ],
+        'Http\Client\HttpClient' => [
+            ['class' => Guzzle6::class, 'condition' => Guzzle6::class],
+            ['class' => Guzzle5::class, 'condition' => Guzzle5::class],
+            ['class' => Curl::class, 'condition' => Curl::class],
+            ['class' => Socket::class, 'condition' => Socket::class],
+            ['class' => Buzz::class, 'condition' => Buzz::class],
+            ['class' => React::class, 'condition' => React::class],
```

Comment:
That is the beauty of Foo::class, it doesn't throw an exception if the class not exists. =)